kernel/rros/thread.rs

```diff
@@ -2543,11 +2506,13 @@ pub fn rros_set_period(
     // if (period < rros_get_clock_gravity(clock, kernel))
     //     return -EINVAL;
 
-    let flags = raw_spin_lock_irqsave();
-    // raw_spin_lock_irqsave(&curr->lock, flags);
+    let flags = thread.irq_lock_noguard();
```

Comment:
I notice that currently the `lock` is manully set and unset. It's nice to use a guard for this in the further PR. You can add a `todo` in our project.